code/_core/obj/item/fishing/rod.dm

```diff
@@ -0,0 +1,282 @@
+/obj/item/fishing/rod
+	name = "fishing rod"
+	drop_sound = 'sound/items/drop/scrap.ogg'
+	size = SIZE_2
+	weight = 2
+	var/lavaproof = 0
+	var/lcolor = "#FFFFFF"
+	var/nice_bait = 0
+	var/break_on_failure = 0
+	var/break_on_success = 0
+	var/win = 0
+	var/win_time = 0
+	var/timemin = 3
+	var/timemax = 9
+	var/react = 9
+	var/bobsprite = null
+	var/catchsound = "sound/ui/message_ping.ogg"
+	var/fishing = 0
+	var/fishingzone = 0
+	var/wished_fishingzone = 0
+	var/mintime = 0
+	var/maxtime = 0
+	var/list/obj/item/effrewards = list()
+	var/list/obj/item/wekrewards = list()
+	var/fish_loc
+
+	var/obj/item/fishing/lure
+	var/obj/item/fishing/line
+	var/obj/item/fishing/bait
+
+/obj/item/fishing/rod/save_item_data(var/save_inventory = TRUE)
+	. = ..()
+	SAVEATOM("lure")
+	SAVEATOM("line")
+	SAVEATOM("bait")
+	return .
+
+/obj/item/fishing/rod/load_item_data_pre(var/mob/living/advanced/player/P,var/list/object_data)
+	. = ..()
+	LOADATOM("lure")
+	LOADATOM("line")
+	LOADATOM("bait")
+	update_rod_stats()
+	return .
+
+/obj/item/fishing/rod/proc/get_line()
+	return line
+
+/obj/item/fishing/rod/proc/get_lure()
+	return lure
+
+/obj/item/fishing/rod/proc/get_bait()
+	return bait
+
+/obj/item/fishing/rod/proc/update_rod_stats()
+	if(line)
+		var/obj/item/fishing/line/L = get_line()
+		lavaproof = L.lavaproof
+		break_on_failure = L.break_on_failure
+		break_on_success = L.break_on_success
+		lcolor = L.color
+	if(lure)
+		var/obj/item/fishing/lure/L = get_lure()
+		bobsprite = L.bobsprite
+		catchsound = L.catchsound
+	if(bait)
+		var/obj/item/fishing/bait/L = get_bait()
+		effrewards = L.effrewards
+		wekrewards = L.wekrewards
+		nice_bait = L.nice_bait
+		wished_fishingzone = L.wished_fishingzone
+	update_overlays()
+	update_sprite()
+	return
+
+/obj/item/fishing/rod/update_overlays()
+	. = ..()
+	if(line)
+		var/image/I = new/image('icons/obj/item/fishing/rod.dmi',"string")
+		I.color = lcolor
+		add_overlay(I)
+	return .
+
+
+/obj/item/fishing/rod/clicked_on_by_object(var/mob/caller,var/atom/object,location,control,params)
+
+	object = object.defer_click_on_object(location,control,params)
+	if(fishing == 0)
+		if(bait && is_inventory(object))

```

Comment:
These is_inventory(object) checks are redundant. You should put is_inventory() on the same line as !fishing

Reply:
doing so makes it unable to insert items again, not quite sure how to fix this one